src/libltfs/ltfs_internal.c

```diff
@@ -982,6 +985,48 @@ int _ltfs_make_lost_found(tape_block_t ip_eod, tape_block_t dp_eod,
 	return 0;
 }
 
+/**
+ * Find the appropriate append address for a partition at which calling ltfs_write_index() will
+ * correctly restore consistency to the partition.
+ *
+ * @param dev The ltfs device to work on
+ * @param part The partition to examine
+ * @param block The starting block of the final index on the partition
+ * @return 0 if eod is the appropriate append address, >0 for the absolute block address to append
+ *         at, <0 on error
+ */
+int _ltfs_find_append_blk_after_idx(struct device_data *dev, tape_partition_t part, tape_block_t block) {
```

Comment:
The cam driver won't do that, it only appends the fm on close if a write has been done since the last fm. Now that's not to say that another LTFS implementation can't do this, but an empty index violates the specification and at that point I think it is fair for ltfs_check_medium to not be able to automatically repair the partition.